odoo/addons/base/models/ir_actions.py

```diff
@@ -592,9 +593,17 @@ def eval_value(self, eval_context=None):
             expr = line.value
             if line.type == 'equation':
                 expr = safe_eval(line.value, eval_context)
-            elif line.col1.ttype in ['many2one', 'integer']:
+            else:
                 try:
-                    expr = int(line.value)
+                    val = int(line.value)
+                    if line.col1.ttype in ['many2one', 'integer'] or line.type == 'key':
+                        expr = val
+                except ValueError:
```

Comment:
What's the purpose of this ?

Reply:
The fallback shouldn't be done in the context of the exception.

Everything should work (with ``if`` en zo voort) without using an additional except.